src/bci_build/package/gcc/README.md.j2

```diff
@@ -0,0 +1,95 @@
+# The GNU Compiler Collection Container Image
+
+The GNU Compiler Collection (GCC) is an optimizing compiler for various
+architectures and operating systems. It is the default compiler of the GNU
+project and the default compiler for most Linux Distributions including SUSE
+Linux Enterprise and openSUSE.
+
+
+## How to use this image
+
+### Compile your application in a `Dockerfile`
+
+Generally you'd want to compile your application and distribute it as part of a
+container image of your own. For that, create a `Dockerfile` as follows which
+builds a container image based on this image, copies the sources to a working
+directory and compiles the application:
+{%- set reference_no_release = (image.reference | replace("-%RELEASE%", "") ) %}
+
+```Dockerfile
+FROM {{ reference_no_release }}
+WORKDIR /src/
+COPY . /src/
+RUN gcc main.c src1.c src2.c
+CMD ["./a.out"]
+```
+
+It is also possible to compile a static binary with gcc as part of a multistage
+build:
+
+```Dockerfile
+FROM {{ reference_no_release }} as builder
+WORKDIR /src/
+COPY . /src/
+RUN gcc -static main.c src1.c src2.c
+
+FROM registry.suse.com/bci/bci-micro:latest
+WORKDIR /build/
+COPY --from=builder /src/a.out /build/app
+CMD ["/build/app"]
+```
+
+Note that you must build a static binary to deploy it into bci-micro as
+otherwise shared libraries might be missing. You cannot deploy such an app into
+a `scratch` image as it is not possible to statically link glibc.
+
+
+### Available build systems
+
+The Container Image comes preloaded with `make` by default. Other build systems
+and related utilities are available in the repository and can be installed via
+`zypper`, including:
+- `meson`
+- `cmake`
+- `ninja`
+- `autoconf` & `automake`
+
+
+### Available compiler frontends
+
+The GNU Compiler Collections supports a multitude of frontends. The container
+image ships the C and C++ frontends available as `gcc` and `g++`
+respectively. The following additional frontends can be installed from the
+repository:
+- `gcc{{ image.version }}-fortran` for Fortran support
```

Comment:
This seems like a maintenance nightmare once versions change and more features are added or removed.

Can we point to our GCC documentation for each release? Something like this https://documentation.suse.com/sbp/server-linux/html/SBP-GCC-12/index.html or another more appropriate if one exists.

Reply:
> This seems like a maintenance nightmare once versions change and more features are added or removed.

Imho it's really only a mess for Tumbleweed. For SLE we have only Fortran & Ada. Since we don't show the readme for Tumbleweed anywhere, we could also just remove the TW specific part.

> Can we point to our GCC documentation for each release? Something like this https://documentation.suse.com/sbp/server-linux/html/SBP-GCC-12/index.html or another more appropriate if one exists.

Unfortunately we do not have a generic manual, nor one for gcc 13.